cpp/src/slice2java/Gen.cpp

```diff
@@ -4243,7 +4243,8 @@ Slice::Gen::ProxyVisitor::visitInterfaceDefEnd(const InterfaceDefPtr& p)
     out << nl << "public static " << prxName
         << " createProxy(com.zeroc.Ice.Communicator communicator, String proxyString)";
     out << sb;
-    out << nl << "return uncheckedCast(communicator.stringToProxy(proxyString));";
+    out << nl << "var ref = communicator.getInstance().referenceFactory().create(proxyString, null);";
+    out << nl << "return (ref == null) ? null : new " << prxIName << "(ref);";
```

Comment:
createProxy is akin to a constructor and should never return null. If you pass an empty proxyString, you should get an exception.

See for example the C# code.

Reply:
Fixed to throw instead of returning null.